tests/testsuite/cargo_add/no_public/mod.rs

```diff
@@ -0,0 +1,39 @@
+use cargo_test_support::compare::assert_ui;
+use cargo_test_support::prelude::*;
+use cargo_test_support::Project;
+
+use cargo_test_support::curr_dir;
+
+#[cargo_test]
+fn case() {
+    cargo_test_support::registry::init();
+    for name in ["my-package1", "my-package2"] {
```

Comment:
As these are new tests, lets make sure we only add the minimum dep/version combinations needed as these slow down test running and move us closer to running out of space in CI (a very rough pass was previously done to minimize these when the impact was realized but more cleanup is possible)

Reply:
See https://github.com/rust-lang/cargo/pull/13046/commits/07ba57dd8ba2a4c4f0e4aac8198a3fc69b48e07b , Only use one package one version for `cargo add`